### PR TITLE
changing error message and error code

### PR DESCRIPTION
**Pull request by dhrubo-os**
### Description
[changing error message and error code. Currently error message is default to the 500 error which is not ideal. So we need to set the correct error which we are receiving from the downstream. ]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dbwiddis — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java [hunk 1]
```diff
@@ -139,7 +141,10 @@ public void onResponse(MLModel mlModel) {
                             }
                         }, e -> {
                             log.error("Failed to Validate Access for ModelId " + modelId, e);
-                            wrappedListener.onFailure(e);
+                            wrappedListener
+                                .onFailure(
+                                    new OpenSearchStatusException("Failed to Validate Access for ModelId " + modelId, RestStatus.FORBIDDEN)
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java:146]: The error message means this is some internal error. Before validating access successfully, how can we know this is forbidden ?

Comment (dhrubo-os) [reply plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java:146]: Good point, but I'm not sure if `INTERNAL_SERVER_ERROR` will be applicable here too. 

Here it's invoking [validateModelGroupAccess](https://github.com/opensearch-project/ml-commons/blob/8ae85a37d608739e57310bd7b9851c41d26bb06a/plugin/src/main/java/org/opensearch/ml/helper/ModelAccessControlHelper.java#L77-L140) method from where I think various kind of exception (Exception, MLResourceNotFoundException, MLValidationException) can be sent.

Do you have any suggestion?

Comment (dbwiddis) [reply plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java:146]: > various kind of exception (Exception, MLResourceNotFoundException, MLValidationException) can be sent.
> 
> Do you have any suggestion?

You could do a text search of the message for the exception name and/or if there is a "Status:" field in the message.

**Comment by samuel-oci** [comment]
@ylwu-amzn can you please make sure to fill in the fields in the PR description? e.g. __Description__, __Issues resolved__ etc..

**Review by samuel-oci — DISMISSED** [review]

**Review by rbhavna — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java (hunk 1 above)
Comment (dhrubo-os) [reply plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java]: @dbwiddis Could you please take a look at the current commit?

**Comment by dhrubo-os** [comment]
> @dhrubo-os can you please make sure to fill in the fields in the PR description? e.g. **Description**, **Issues resolved** etc..

Added description.

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**